src/pages/home/uploadSong/AddOwnerModal.tsx

```diff
@@ -1,17 +1,49 @@
+import * as Yup from "yup";
 import { FunctionComponent } from "react";
-import { Box, DialogProps, Stack, Typography, useTheme } from "@mui/material";
+import { useSelector } from "react-redux";
+import { Form, Formik, FormikValues } from "formik";
+import { Box, DialogProps, Stack, Typography } from "@mui/material";
 import { Button, Dialog, HorizontalLine } from "elements";
-import { SwitchField, TextInputField } from "components";
+import { DropdownSelectField, SwitchField, TextInputField } from "components";
+import { commonYupValidation } from "common";
+import { selectContent } from "modules/content";
+import theme from "theme";
 
 interface AddOwnerModalProps extends Omit<DialogProps, "onClose"> {
   readonly onClose: VoidFunction;
+  readonly onSubmit: (values: FormikValues) => void;
 }
 
 const AddOwnerModal: FunctionComponent<AddOwnerModalProps> = ({
   open,
   onClose,
+  onSubmit,
 }) => {
-  const theme = useTheme();
+  const { roles } = useSelector(selectContent);
+
+  const initialValues = {
+    email: "",
+    firstName: "",
+    isCreator: false,
+    isRightsOwner: false,
+    lastName: "",
+    role: "",
+  };
+
+  const validationSchema = Yup.object().shape({
+    email: commonYupValidation.email,
+    firstName: commonYupValidation.firstName,
+    isCreator: Yup.boolean()
+      .required()
+      .test(
+        "at-least-one-true",
+        "Rights or Credits must be true",
+        (item, testContext) => item || testContext.parent.isRightsOwner
+      ),
+    isRightsOwner: Yup.boolean(),
+    lastName: commonYupValidation.lastName,
+    role: commonYupValidation.role(roles),
+  });
```

Comment:
I also agree with part of the slack discussions, the `Add` button should be disabled if one of the options requires marking to move on. My opinion, and not a code change requirement at this stage but should confer with design, is that the text should probably still be shown when hovering over the `Add` button or in some other type of info tip since design didn't agree with the included one. Might not be an issue though as design said in the users point of view and have to see how the user input goes.

Reply:
Yea i went with the error message because that's what we had done for the sale page. seems like design wants something different now(which is my preferable solution). I will be making the change to delete the error text and disable the button

Reply:
Disabling the submit button until all required fields are completed is a better option imo, compared to enabling the submit button and showing error messages when the user attempts to submit. Disabling the submit button provides a clear indication of required fields and prevents incomplete submissions, helping to avoid frustration and confusion. On the other hand, enabling the submit button and showing error messages only when clicked can result in confusion and missed error messages, leading to incomplete submissions

Reply:
Agree with Trevor, and on board now with not disabling the button in this scenario and any form based settings. I believe the red warning borders and messages provide enough context for the user, especially since they won't be able to move forward without completing the indicated fields, I wouldn't say any of those errors could be missed in that scenario. If needed can bring up to design team if a polite toast message or informational bubble pops up saying please fill in the fields. As long as there's enough context for the user to understand what needs to be submitted than I think we are winning. Also agree on disabling in the "consent" scenario.

In this scenario I believe it's also important to properly notify the user of those two trigger option requirements, maybe a red box around both signifying a need for selection, I just don't think the words underneath the two boxes does it justice. Either way we should be consistent with types of user warnings we give for each input field type.

Bit of a rabbit hole, but my thought is that this modal or minting in general might require some level of first time tour or video informational clues because minting is a big deal and understanding all these pieces will be important for the user as well, to give a little more context of the page and not just hey! a field missing type of warning. Future Post MVP possibly.

Reply:
From the convo above, i will be leaving it as is, in this case it will be enabled by default and then display the errors.